sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridClient.cs

```diff
@@ -163,5 +187,39 @@ public virtual Response PublishCustomEvents(IEnumerable<object> events, Cancella
                 throw;
             }
         }
+
+        /// <summary>
+        /// Creates a SAS token for use with Event Grid service
+        /// </summary>
+        /// <param name="expirationUtc">Expiration time</param>
+        /// <returns></returns>
+        public string BuildSharedAccessSignature(DateTimeOffset expirationUtc)
+        {
+            const char Resource = 'r';
+            const char Expiration = 'e';
+            const char Signature = 's';
+
+            if (_key == null)
+            {
+                throw new NotSupportedException();
```

Comment:
nit: we should include a message in the exception that explains this method can only be used when using an EventGridClient created using AzureKeyCredential.